skimage/feature/tests/test_peak.py

```diff
@@ -351,57 +355,35 @@ def test_four_quadrants(self):
                                          exclude_border=False)
         assert np.all(result == expected)
 
-    def test_disk(self):
-        '''regression test of img-1194, footprint = [1]
-        Test peak.peak_local_max when every point is a local maximum
-        '''
-        image = np.random.uniform(size=(10, 20))
-        footprint = np.array([[1]])
-        with expected_warnings(["indices argument is deprecated",
-                                "When footprint.size < 2"]):
-            result = peak.peak_local_max(image, labels=np.ones((10, 20)),
-                                         footprint=footprint,
-                                         min_distance=1, threshold_rel=0,
-                                         threshold_abs=-1, indices=False,
-                                         exclude_border=False)
-        assert np.all(result)
-        with expected_warnings(["indices argument is deprecated",
-                                "When footprint.size < 2"]):
-            result = peak.peak_local_max(image, footprint=footprint,
-                                         threshold_abs=-1,
-                                         indices=False,
-                                         exclude_border=False)
-        assert np.all(result)
-
     def test_3D(self):
         image = np.zeros((30, 30, 30))
-        image[15, 15, 15] = 1
+        image[16, 16, 16] = 1
```

Comment:
Why did you change this?

Reply:
Previously for a peak `p_0`, any other peak `p_1` verifying

```
d(p_0, p_1) < min_distance
```
was removed.
In the current implementation the condition became

```
d(p_0, p_1) <= min_distance
```
IMHO, it corresponds more to `peak_local_max` doc statement
> Peaks are the local maxima in a region of 2 * min_distance + 1 (i.e. peaks are separated by at least min_distance).

Reply:
I don't think that's right? I would say that `at least x` means, `x or more`. So peaks separated by exactly x should remain.

Reply:
:facepalm:... You're right! This must be fixed!

Reply:
I am facing a dilemma @jni :confused: . I fixed this behavior, and when I ran the tests I realized that with this convention `min_distance=1` is equivalent to `min_distance=0`!